src/cc/libbpf.c

```diff
@@ -782,6 +785,14 @@ static int bpf_try_perf_event_open_with_probe(const char *name, uint64_t offs,
   if (is_return)
     attr.config |= 1 << is_return_bit;
 
+  // Per bcc#2230, a6ca88b introduces a ref_ctr_offset to uprobes to set
+  // semaphores directly through uprobe call, rather than seek to memory
+  // the ref_ctr_offset must be packed in the top 32 bits of the 64 bit storage
+  // this supports incrementing the reference counter directly from mm_struct
+  // rather than seeking /proc/PID/mem in userspace
+  if (ref_ctr_offset > 0)
+    attr.config |= (uint64_t)ref_ctr_offset << PERF_UPROBE_REF_CTR_OFFSET_SHIFT;
```

Comment:
Since the offset is only 32 bits, I decided to preserve this type information. To avoid overflowing it on the bitshift before it is upcast to 64 bits, I explicitly cast it to a 64 bit container.